login-fire-form.html

```diff
@@ -441,12 +455,14 @@
       this._cleanMessages();
       e = this.$$('#emailInput').validate()
       if (e) {
+        this._inProgress = true;
         this.$$('#auth').auth.sendPasswordResetEmail(this.email)
           .then(function() {
-            this.set('_lastMessage',
-              this.localize('lf-resetpw-success-message'));
+            this.set('_lastMessage', this.localize('lf-resetpw-success-message'));
+            this._inProgress = false;
           }.bind(this), function(error) {
             this.set('_lastError', error.message);
+            this._inProgress = false;
           }.bind(this));
```

Comment:
Maybe it's possible to replace the "new" lines 462 and 465 by a call to the [Promise.prototype.finally](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Promise/finally) method on the response of the `sendPasswordResetEmail()`?

Something like:
```javascript
this.$$('#auth').auth.sendPasswordResetEmail(this.email)
	.then(function(resp){/*...*/}.bind(this), function(error){/*...*/}.bind(this))
	.finally(function(){/*...*/ this._inProgress = false;}.bind(this));
```

Reply:
Yep. Still new to Promise land. :grin: Will change that one as well.

Reply:
I took a look at your repos and I can guess what you're made of 😉 !

🆒 Keep Calm & Contribute 😎